### PR TITLE
poc: many strategies pagination

### PR DESCRIPTION
**Pull request by ivarconr**
This fixes the case when a customer have thousands of strategies causing the react UI to crash. We still consider it incorrect to use that amount of strategies and this is more a workaround to help the customer out of a crashing state.

We put it behind a flag called `manyStrategiesPagination` and plan to only enable it for the customer in trouble.

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyNonDraggableItem.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+interface IStrategyItemProps {
+    strategy: IFeatureStrategy;
+    environmentName: string;
+    index: number;
+    otherEnvironments?: IFeatureEnvironment['name'][];
+}
+
+/**
+ * @deprecated
+ */
```
Comment (ivarconr) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyNonDraggableItem.tsx:30]: I decided to just copy the `StrategyDraggableItem` to remove the draggable support. Did not want to pollute the exiting component with more options.

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx [hunk 2]
```diff
@@ -50,9 +55,12 @@ const EnvironmentAccordionBody = ({
         usePendingChangeRequests(projectId);
     const { setToastData, setToastApiError } = useToast();
     const { refetchFeature } = useFeature(projectId, featureId);
+    const manyStrategiesPagination = useUiFlag('manyStrategiesPagination');
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx:58]: Oh shoot.

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx [hunk 3]
```diff
@@ -63,10 +71,20 @@ const EnvironmentAccordionBody = ({
         setStrategies(featureEnvironment?.strategies || []);
     }, [featureEnvironment?.strategies]);
 
+    useEffect(() => {
+        if (strategies.length > 50) {
+            trackEvent('many-strategies');
+        }
+    }, []);
```
Comment (ivarconr) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx:78]: was this correct way to add tracking @kwasniew ?

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx [hunk 4]
```diff
@@ -195,21 +213,68 @@ const EnvironmentAccordionBody = ({
...
-                                    otherEnvironments={otherEnvironments}
-                                    isDragging={dragItem?.id === strategy.id}
-                                    onDragStartRef={onDragStartRef}
-                                    onDragOver={onDragOver(strategy.id)}
-                                    onDragEnd={onDragEnd}
-                                />
-                            ))}
-                        </>
+                        <ConditionallyRender
+                            condition={
```
Comment (ivarconr) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx:217]: We want to show the normal behavior when strategies are fewer than 50 and when the flag is disabled.

File: frontend/src/hooks/usePagination.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ const usePagination = <T>(
         }
 
         const result = paginate(dataToPaginate, limit);
-        setPageIndex(0);
+        // setPageIndex(0);
```
Comment (sighphyre) [inline frontend/src/hooks/usePagination.ts:23]: This looks very subtle, what else does this effect?

Comment (ivarconr) [reply frontend/src/hooks/usePagination.ts:23]: hopefully nothing. I could not find any other usage of this hook. It solved all my pagination needs so I decided to use it.

Comment (sighphyre) [reply frontend/src/hooks/usePagination.ts:23]: Okay deprecated hook. Doesn't seem to be in use anywhere

**Review by sighphyre — APPROVED** [review]
Sure, I think this is okay for an interim fix. My knowledge of this portion of the front-end isn't solid but I think the approach is correct

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx (hunk 3 above)
Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/EnvironmentAccordionBody.tsx:78]: seems to be correcy

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyNonDraggableItem.tsx (hunk 1 above)
Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyNonDraggableItem.tsx:30]: What about using existing component and passing no-op functions to drag properties?

Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyNonDraggableItem.tsx:30]:
```
<StrategyDraggableItem
                                    key={strategy.id}
                                    strategy={strategy}
                                    index={index}
                                    environmentName={featureEnvironment.name}
                                    otherEnvironments={otherEnvironments}
                                    isDragging={false}
                                    onDragStartRef={(() => {}) as any}
                                    onDragOver={(() => {}) as any}
                                    onDragEnd={(() => {}) as any}
                                />
```

**Pull request merged**